src/trix/models/document.coffee

```diff
@@ -219,9 +219,11 @@ class Trix.Document extends Trix.Object
   applyBlockAttributeAtRange: (attributeName, value, range) ->
     {document, range} = @expandRangeToLineBreaksAndSplitBlocks(range)
 
-    if Trix.config.blockAttributes[attributeName].listAttribute
+    if getBlockAttributes()[attributeName].listAttribute
       document = document.removeLastListAttributeAtRange(range, exceptAttributeName: attributeName)
       {document, range} = document.convertLineBreaksToBlockBreaksInRange(range)
+    else if getBlockAttributes()[attributeName].terminal
+      {document, range} = document.convertLineBreaksToBlockBreaksInRange(range)
```

Comment:
`removeLastListAttribute` (`document = document.removeLastListAttributeAtRange(range, exceptAttributeName: attributeName`) should not apply to terminal blocks, so I don't think these can be combined. I suppose they could in a nested if statement and it'd be more DRY, but is not as clear about what is happening:

```coffeescript
config = getBlockAttributes()[attributeName]

if config.listAttribute or config.terminal
  if config.listAttribute
    document = document.removeLastListAttributeAtRange(range, exceptAttributeName: attributeName)
  {document, range} = document.convertLineBreaksToBlockBreaksInRange(range)
else
  ...
```

I like introducing a local variable here for the attribute though.